core/shared/src/main/scala/zio/logging/LogFilter.scala

```diff
@@ -371,7 +400,12 @@ object LogFilter {
         case (xFirst :: xTail, yFirst :: yTail) =>
           val r = yFirst.compareTo(xFirst)
           if (r != 0) {
-            r
+            if (xFirst.contains('*') || yFirst.contains('*')) {
+              if (Set("**", "*").contains(xFirst)) 1
+              else if (Set("**", "*").contains(yFirst)) -1
+              else
+                compareNames(xFirst.split('*').toList.filter(_.nonEmpty), yFirst.split('*').toList.filter(_.nonEmpty))
+            } else r
```

Comment:
LogFilterSpec.nameLevelOrdering should be extended to check proper ordering ...

Reply:
extended the spec and resolved an ordering issue between `*` vs `**`